### DISCUSSION TITLE
I am getting the error while creating selectFilter using relstionships

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#9138] [Help]
I want to create a selectFilter for user age on select of age it should filter the data 

Below is my code,SelectFilter::make('age')
                ->relationship('personalInfo','age')
                ->options([
                    '45'       => '45',
                    '50'   => '50',
                    '55'     => '55',
                    '60'      => '60',
                    '70'         => '70',
                ])
                // ->label('Age Filter')
                ]) I am getting the below error,
        Filament\Forms\Components\Select::isOptionDisabled(): Argument #2 ($label) must be of type string, null given, called in C:\xampp\htdocs\EldersTalentNew\storage\framework\views\7f8ee5cc64e0b96f483b61bf3693ee53.php on line 86

 I want to make range filter for ages  my filament version is 3

**Comment by leandrocfe** [comment]
Probably a relationship issue because this should work
```php
SelectFilter::make('age')
->relationship('personalInfo','age')
```

If you don't want to use the relationship, you can do this

```php
SelectFilter::make('your_column_name')
    ->options([
        '45' => '45',
        '50' => '50',
        '55' => '55',
        '60' => '60',
        '70' => '70',
    ])
    ->label('Age Filter')
```

**Reply by mohandtharsha** [reply]
after above changes I am getting the below error,

Filament\Forms\Components\Select::isOptionDisabled(): Argument #2 ($label) must be of type string, null given, called in C:\xampp\htdocs\EldersTalentNew\storage\framework\views\7f8ee5cc64e0b96f483b61bf3693ee53.php on line 86

Below is my relationship,
public function personalInfo()
    {
        return $this->belongsTo(PersonalInfo::class, 'user_id', 'user_id');
    } I want to make range filter for age

**Reply by leandrocfe** [reply]
Are you sure the relationship is working? I think it is returning null...

**Reply by leandrocfe** [reply]
Maybe some age value is `null` instead of `''`

**Reply by leandrocfe** [reply]
```php
SelectFilter::make('age')
->relationship('personalInfo','age', modifyQueryUsing: fn (Builder $query) => $query->whereNotNull('age'))
```

**Comment by adiputra22** [comment]
Anyone have same face this issue? 

```
SelectFilter::make('category')
      ->label('Category')
      ->options(function () {
          $categories = $this->parent->ingredients()
              ->pluck('category')
              ->unique()
              ->toArray();

          return array_combine($categories, $categories);
      })
      ->query(function (Builder $query, array $data) {
          if (! empty($data['value'])) {
              $query->whereHas(
                  'ingredient',
                  fn (Builder $query) => $query->where('category', '=', $data['value'])
              );
          }
      }),
```

the issue is same message

```
Illuminate\View\ViewException · Filament\Forms\Components\Select::isOptionDisabled(): Argument #2 ($label) must be of type string, null given, called in /home/forge/xxx/storage/framework/views/d105b1c0ab8d8c2f7f77fc7d245fa4a1.php on line 97 (View:
```

![Screen Shot 2024-07-04 at 09 45 37](https://github.com/filamentphp/filament/assets/16931399/27bd0a08-82a4-4cd8-b8a8-70f026f2be77)


I do test on local with same php version it working fine even the category is empty, but In production, It throwing error like that.
I not sure how to test because I not able to manage production server.
